src/auth/jwtaccess.ts

```diff
@@ -51,13 +58,17 @@ export class JWTAccess {
   }
 
   /**
-   * Get a non-expired access token, after refreshing if necessary
+   * Get a non-expired access token, after refreshing if necessary.
    *
    * @param {string} authURI the URI being authorized
-   * @param {function} metadataCb a callback invoked with the jwt request metadata.
-   * @returns a Promise that resolves with the request metadata response
+   * @returns An object that includes the authorization header.
    */
   getRequestMetadata(authURI: string): RequestMetadataResponse {
+    this.cleanupCache();
```

Comment:
This seems unnecessary, just check the expiration on the key you get from the cache and drop it if needed. O(1) vs O(n).

(This makes me yearn for Node.js to have a LRU Cache in it's "standard library")

Reply:
The danger there is that the cache can grow over time to a large size, leading to leaks.  I am taking a shot at using the canonical LRU cache module.